source/h1_encoder.c

```diff
@@ -52,18 +53,41 @@ static int s_scan_outgoing_headers(
                 }
             } break;
             case AWS_HTTP_HEADER_CONTENT_LENGTH: {
+                has_content_length_header = true;
                 struct aws_byte_cursor trimmed_value = aws_strutil_trim_http_whitespace(header.value);
                 if (aws_strutil_read_unsigned_num(trimmed_value, &encoder_message->content_length)) {
                     AWS_LOGF_ERROR(AWS_LS_HTTP_STREAM, "id=static: Invalid Content-Length");
                     return aws_raise_error(AWS_ERROR_HTTP_INVALID_HEADER_VALUE);
                 }
-                if (encoder_message->content_length > 0) {
-                    has_body_headers = true;
+            } break;
+            case AWS_HTTP_HEADER_TRANSFER_ENCODING: {
+                has_transfer_encoding_header = true;
+                if (0 == header.value.len) {
+                    AWS_LOGF_ERROR(AWS_LS_HTTP_STREAM, "id=static: Transfer-Encoding must include a valid value");
+                    return aws_raise_error(AWS_ERROR_HTTP_INVALID_HEADER_VALUE);
+                }
+                struct aws_byte_cursor chunked = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("chunked");
```

Comment:
not sure why we don't hoist that up out of the function scope. Compiler prolly doesn't care but seems cleaner.

Reply:
made this look like the other equality checks in the file